src/policies/cooler/CoolerV2Migrator.sol

```diff
@@ -0,0 +1,439 @@
+// SPDX-License-Identifier: GLP-3.0
+pragma solidity ^0.8.15;
+
+// Interfaces
+import {IERC3156FlashBorrower} from "src/interfaces/maker-dao/IERC3156FlashBorrower.sol";
+import {IERC3156FlashLender} from "src/interfaces/maker-dao/IERC3156FlashLender.sol";
+import {IDaiUsdsMigrator} from "src/interfaces/maker-dao/IDaiUsdsMigrator.sol";
+import {ICoolerV2Migrator} from "../interfaces/cooler/ICoolerV2Migrator.sol";
+
+// Libraries
+import {ReentrancyGuard} from "solmate/utils/ReentrancyGuard.sol";
+import {SafeCast} from "src/libraries/SafeCast.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/tokens/ERC20.sol";
+
+// Bophades
+import {Kernel, Keycode, Permissions, Policy, toKeycode} from "src/Kernel.sol";
+import {CoolerFactory} from "src/external/cooler/CoolerFactory.sol";
+import {Clearinghouse} from "src/policies/Clearinghouse.sol";
+import {Cooler} from "src/external/cooler/Cooler.sol";
+import {CHREGv1} from "src/modules/CHREG/CHREG.v1.sol";
+import {RGSTYv1} from "src/modules/RGSTY/RGSTY.v1.sol";
+import {IMonoCooler} from "src/policies/interfaces/cooler/IMonoCooler.sol";
+import {IDLGTEv1} from "src/modules/DLGTE/IDLGTE.v1.sol";
+import {ROLESv1} from "src/modules/ROLES/ROLES.v1.sol";
+import {PolicyEnabler} from "src/policies/utils/PolicyEnabler.sol";
+
+/// @title  CoolerV2Migrator
+/// @notice A contract that migrates debt from Olympus Cooler V1 facilities to Cooler V2.
+///         This is compatible with all three versions of Cooler V1.
+/// @dev    This contract uses the `IERC3156FlashBorrower` interface to interact with Maker flashloans.
+contract CoolerV2Migrator is
+    IERC3156FlashBorrower,
+    ICoolerV2Migrator,
+    ReentrancyGuard,
+    Policy,
+    PolicyEnabler
```

Comment:
Why does this need to be a policy out of interest? I first thought it would just be something external

Reply:
Bad experience with CoolerUtils (first version of the LoanConsolidator) - it wasn't a policy and couldn't be disabled after there was an issue found. This is a policy so that it can be enabled/disabled by a permissioned user. If it was an external contract, it would need an owner (e.g. emergency MS). If that emergency MS address was ever changed, this contract would also need to be updated with the new owner, and being outside of the protocol stack, it'd likely get forgotten.

I'm open to suggestions about how to avoid it being a policy but still have enable/disable functionality and ownership.